src/fe/fe.C

```diff
@@ -306,8 +306,21 @@ void FE<Dim,T>::reinit(const Elem * elem,
         // appropriate. We're just gonna have to trust the user on this one. If
         // they "screw up" we'll throw an exception from the LU decomposition,
         // and they can choose to handle it or not
-        this->compute_dual_shape_coeffs();
 
+        // instead of using the customized qrule from mortar segments,
+        // we need the default qrule for computing the dual coefficients
+        FEType fe_type(this->get_order(), T);
+        QGauss qrule(elem->dim(), fe_type.default_quadrature_order());
+        // In preparation of computing dual_coeff, we compute the default shape
+        // function values and save these in dual_phi (instead of declaring a
+        // new container). The TRUE dual_phi values are computed in
+        // compute_dual_shape_functions()
+        all_shapes(elem, fe_type.order, qrule.get_points(), this->dual_phi);
+
+        this->compute_dual_shape_coeffs(qrule);
```

Comment:
It would be better if the `compute_dual_shape_coeffs` is only done once and the `dual_coeff` is saved somewhere for further use. Because here, the computed `dual_coeff` should be the same for all elements that are of the same FEFamily and a fixed dimension. This may same some computer time. Any ideas @lindsayad ?

Reply:
This sounds reasonable to me... we cache normal shape functions values in FEs already, so I think if you follow the logic used for that, you should be able to cache dual shape function values as well. As you pointed out, since you don't need derivatives of the dual shape function values, the cache should not become invalidated for different length elements, only different geometric types.

Reply:
> we cache normal shape functions values in FEs already, so I think if you follow the logic used for that, you should be able to cache dual shape function values as well.

Thanks @jwpeterson ! Just to be clear, the `dual_coefficient` (used to calculate the `dual shape function`) is constant for a certain FEFamily. However, the `dual shape function` value and its derivatives change along with the `shape function values` (due to the mortar segment update). So I would like to cache the `dual_coefficient`, not the `dual shape function` or its derivatives

Following how shape function values are cached sounds reasonable. Not sure how to do that now, let me look into it a bit more